src/utils/application.data.ts

```diff
@@ -88,18 +99,42 @@ export const deleteApplicationData = (session: Session | undefined): boolean | u
   return session?.deleteExtraData(APPLICATION_DATA_KEY);
 };
 
-export const setApplicationData = async (session: Session | undefined, data: any, key: string): Promise<undefined | void> => {
+export const setApplicationData = async (sessionOrRequest: Request | Session | undefined, data: any, key: string): Promise<undefined | void> => {
 
-  let appData: ApplicationData = await getApplicationData(session);
+  let appData: ApplicationData;
+  let req: Request | undefined;
+  let session: Session | undefined;
 
-  if (ApplicationDataArrayType.includes(key)){
-    if ( !appData[key] ) { appData[key] = []; }
-    appData[key].push(data);
-  } else {
-    appData = { ...appData, [key]: { ...data } } as ApplicationData;
+  try {
+
+    if (!isActiveFeature(FEATURE_FLAG_ENABLE_REDIS_REMOVAL) || (sessionOrRequest instanceof Session)) {
```

Comment:
i think we might not need the flag check here, getApplicationData can take either session or request and we seem to call the same function in the if and else. Can't we just call getApplicationData with sessionOrRequest ?

Reply:
ok, we can leave it like this.

Reply:
cheers